build.sh

```diff
@@ -0,0 +1,166 @@
+  echo "-c: Build command, [ update | download | build | sign | upload ]"
+  echo "-d: Enable bash debug output"
+  echo "-h: Show this help"
+  echo "-m: Optional setting for make options."
+  echo "    Default is \"${MAKEOPTS}\""
+  echo "-n: String for build number is optional"
+  echo "    Default is: \"${BUILD_NUMBER}\""
+  echo "-t: Set gluon targets for build."
+  echo "    Default is: \"${GLUON_TARGETS}\""
+  echo "-w: Path for workspace used for Gluon site directory, e.g. current work directoy"
+  echo "    Default is: \"${GLUON_SITEDIR}\""
+}
+
+# Evaluate arguments for build script.
+if [[ "${#}" == 0 ]]; then
+  usage
+  exit ${E_MISSING_ARG}
+fi
+
+# Evaluate arguments for build script.
+while getopts b:c:dhm:n:t:w: flag; do
+  case ${flag} in
+    b)
+        GIT_BRANCH="${OPTARG}"
+        ;;
+    c)
+      case "${OPTARG}" in
+        update| \
+        download| \
+        build| \
+        sign| \
+        upload)
+          COMMAND="${OPTARG}"
+          ;;
+        *)
+          echo "Error: Invalid build command set."
+          usage
+          exit ${E_WRONG_ARG}
+          ;;
+      esac
+      ;;
+    d)
+      set -x
+      ;;
+    h)
+      usage
+      exit
+      ;;
+    m)
+      MAKEOPTS="${OPTARG}"
+      ;;
+    n)
+      BUILD_NUMBER="${OPTARG}"
+      ;;
+    t)
+      GLUON_TARGETS="${OPTARG}"
+      ;;
+    w)
+      # Use the root project as site-config for make commands below
+      GLUON_SITEDIR="${OPTARG}"
+      ;;
+    *)
+      usage
+      exit ${E_WRONG_ARG}
+      ;;
+  esac
+done
+
+shift $(( OPTIND - 1 ));
+
+if [ -z "${GIT_BRANCH}" ]; then
+  # Set branch name
+  GIT_BRANCH=$(git symbolic-ref -q HEAD)
+  GIT_BRANCH=${GIT_BRANCH##refs/heads/}
+  GIT_BRANCH=${GIT_BRANCH:-HEAD}
+fi
+
+if [ -z "${COMMAND}" ]; then
+  echo "Error: Build command with -c is not set."
+  usage
+  exit ${E_MISSING_ARG}
+fi
+
+# Configure gluon build environment
+export GLUON_BRANCH="${GIT_BRANCH#origin/}"            # Use the current git branch as autoupdate branch
+export GLUON_BRANCH="${GLUON_BRANCH//\//-}"
+export GLUON_BUILD="${BUILD_NUMBER}-$(date '+%Y%m%d')" # ... and generate a fency build identifier
+export GLUON_RELEASE="${GLUON_BRANCH}-${GLUON_BUILD}"
+export GLUON_PRIORITY=1                                # Number of days that may pass between releasing an updating
+export GLUON_SITEDIR
+
+update() {
+  make update ${MAKEOPTS}
+  for GLUON_TARGET in ${GLUON_TARGETS}; do
+    echo "--- Update Gluon for target: ${GLUON_TARGET}"
+    make clean ${MAKEOPTS} GLUON_TARGET=${GLUON_TARGET}
+  done
+}
+
+download() {
+  for GLUON_TARGET in ${GLUON_TARGETS}; do
+    echo "--- Download Gluon Dependencies for target: ${GLUON_TARGET}"
+    make download ${MAKEOPTS} GLUON_TARGET=${GLUON_TARGET}
+  done
+}
+
+build() {
+  for GLUON_TARGET in ${GLUON_TARGETS}; do
+    echo "--- Build Gluon Dependencies for target: ${GLUON_TARGET}"
+    make ${MAKEOPTS} GLUON_TARGET=${GLUON_TARGET}
```

Comment:
Falls in einem target ein lehrzeichen vorkommen sollte, geht das schon bei dem loop über die targets kaputt.